include/internal/catch_result_builder.h

```diff
@@ -34,6 +35,26 @@ namespace Catch {
         std::ostringstream oss;
     };
 
+    struct AnyTypeHolderBase {
+        virtual std::string toString() = 0;
+    };
+
+    template <typename T>
+    struct Decay { typedef T type; };
+
```

Comment:
This `Decay` implementation is incomplete. I had problems with two cases: `std::atomic<T>` and function pointers (couldn't compare a function with a function pointer).